### PR TITLE
mock uuid replaced with spyOn

### PR DESCRIPTION
**Pull request by adoroshk**
### Summary
Currently the uuid test mocks are hard coded. In order to ensures that every test is fresh and independent of each other, the proposed solution was to use beforeAll() to set up the mock. As jest.mock can't be used inside of the tests or inside beforeAll, jest.spyOn was used instead of jest.mock.

**What was changed:**

1. Hard-coded jest.mock were replaced with jest.spyOn inside the beforeAll().
2. The mocks were restored in afterAll().
3. The testing section in documentation was updated to reflect the suggested testing method.

### Testing
This change was tested using:

- [x] WDIO
- [x] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews
- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

**This PR is a part of following JIRA:**
UXPLATFORM-9515

### CODE REVIEW

File: packages/terra-menu/src/_MenuContent.jsx [hunk 1]
```diff
@@ -123,6 +121,8 @@ class MenuContent extends React.Component {
     this.validateFocus = this.validateFocus.bind(this);
     this.needsFocus = props.isFocused;
     this.handleContainerRef = this.handleContainerRef.bind(this);
+    this.menuHeaderId = `terra-menu-headertitle-${uuidv4()}`;
```
Comment (adoroshk) [inline packages/terra-menu/src/_MenuContent.jsx:124]: Hard-coded outside of the component `menuHeaderId` and `menuTopHeaderId` were preventing jest.spyOn.mockImplementation work in Menu.test.jest. 
This change is consistent with other terra element's like this: https://github.com/cerner/terra-framework/blob/3ccdc22e9a8a13014688dcf13aa667bda90cff07/packages/terra-date-input/src/DateInput.jsx#L137
This change is not user-facing and doesn't need to be reflected in changelog.

**Comment by adoroshk** [comment]
> Fails
> 🚫 	Please include a CHANGELOG entry for each changed package this PR. Looks like a CHANGELOG entry is missing for:
> 
>     * terra-menu
> 
> 
> Generated by 🚫 [dangerJS](https://danger.systems/js) against [7ddcd1c](https://github.com/cerner/terra-framework/commit/7ddcd1cebad55c58cf264a1e98a35f31f4ae7ce5)

The change in Menu is not user facing and doesn't need a changelog update, more details here: https://github.com/cerner/terra-framework/pull/1765#discussion_r1307914745

**Review by kenk2 — APPROVED** [review]

File: packages/terra-date-input/tests/jest/DateInput.test.jsx [hunk 2]
```diff
@@ -2,14 +2,21 @@ import React from 'react';
...
-
 describe('DateInput', () => {
+  let mockSpyUuid;
+  beforeAll(() => {
+    mockSpyUuid = jest.spyOn(uuidv4, 'v4').mockImplementation(() => '00000000-0000-0000-0000-000000000000');
+  });
+
+  afterAll(() => {
+    mockSpyUuid.mockRestore();
+  });
```
Comment (sdadn) [inline packages/terra-date-input/tests/jest/DateInput.test.jsx:18]: Instead of doing this twice, I believe we can do it once outside the describe block.

Comment (adoroshk) [reply packages/terra-date-input/tests/jest/DateInput.test.jsx]: It works, fixed here: https://github.com/cerner/terra-framework/pull/1765/commits/6711ca76a70529f38ca38ff4fce256a990abb5bb

This commit also replaces `mockImplementation` with `mockReturnValue`

**Review by sdadn — APPROVED** [review]

**Pull request merged**